starlette/routing.py

```diff
@@ -541,17 +544,20 @@ def url_path_for(self, name: str, **path_params: str) -> URLPath:
 
         partial = None
 
+        scope["routes"] = {"full": None, "partial": []}
```

Comment:
I wonder what would be the use case to have `full` or `partial` match in here? I have a use case similar to yours and what I'm after is the path of the route which was matched and used by router - I'm not really interested whether that was a full or partial match.

Reply:
I’m in the same place as you. I only need the full match, but the partial ones might be need by someone for debug purposes!

Reply:
What is you simply add `route_path` which would only contain the path of the selected router regardless whether it was partial or full match?

Reply:
I’m the case of the partial path, there can be multiple paths. By doing this way I think it gives more versatility towards multiple use cases in the future!

Reply:
I think that for both partial and full match the resource name (route path) in the same, according to how it's defined - same `path`, but different `method`, so there should not be multiple route paths.

Having matching routes might be useful though, but it makes it a bit harder to retrieve route path - you need to have enough knowledge about the underlying ASGI framework to do that.

I'm just trying to think what would be a more standard way of doing this across different frameworks. We are using a library which implements tracing middleware for ASGI applications and it's framework agnostic. Right now it has to have knowledge about Starlette in order to retrieve route path, but it would rather not. 

Would be nice if it was part of the ASGI spec :)

Reply:
AFAIK ASGI has no common way of setting this value, is up to the framework to do it. But I might be wrong, is so, please post a reference here so I can read about it! But if that’s the case, we have no better way to do it. And I still think that this way gives more flexibility. Although I would be ok to change it if requested by the owners.

FYI my first solution only added the matched route, this was a requested change so we agree here!